From 86c2945391cc63f1a4aaa8fc3a53c035c39a820b Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Tue, 2 Sep 2025 13:56:58 +0200
Subject: [PATCH] arm64: dts: st: add cap-power-off-card on stm32mp2x DKs

Since [1] (from LTS v6.6.87), and due to the absence of cap-power-off-card
property in wifi sdmmc parent node, the wifi adapter is no more turned off
on suspend and re-probed on resume, causing watchdog.
To avoid this watchdog, add cap-power-off-card property in wifi sdmmc node
of stm32mp215f-dk, stm32mp235f-dk and stm32mp257f-dk. Using this property
turns on runtime PM on sdmmc bus.

[1] commit ec7315e68d88 ("wifi: brcmfmac: keep power during suspend if board requires it")

Change-Id: I40168724871c70f7156cd779ad6be2292bbd4866
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/486711
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Christophe ROULLIER <christophe.roullier@foss.st.com>
Domain-Review: Christophe KERELLO <christophe.kerello@st.com>
Tested-by: Christophe ROULLIER <christophe.roullier@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp215f-dk.dts | 1 +
 arch/arm64/boot/dts/st/stm32mp235f-dk.dts | 1 +
 arch/arm64/boot/dts/st/stm32mp257f-dk.dts | 1 +
 3 files changed, 3 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp215f-dk.dts b/arch/arm64/boot/dts/st/stm32mp215f-dk.dts
index 4a426bc00597..9101022ce967 100644
--- a/arch/arm64/boot/dts/st/stm32mp215f-dk.dts
+++ b/arch/arm64/boot/dts/st/stm32mp215f-dk.dts
@@ -544,6 +544,7 @@ &sdmmc2 {
 	pinctrl-1 = <&sdmmc2_b4_od_pins_a>;
 	pinctrl-2 = <&sdmmc2_b4_sleep_pins_a>;
 	st,neg-edge;
+	cap-power-off-card;
 	cap-sdio-irq;
 	bus-width = <4>;
 	vmmc-supply = <&scmi_v3v3>;
diff --git a/arch/arm64/boot/dts/st/stm32mp235f-dk.dts b/arch/arm64/boot/dts/st/stm32mp235f-dk.dts
index 04a447ceb99a..b705d40d4793 100644
--- a/arch/arm64/boot/dts/st/stm32mp235f-dk.dts
+++ b/arch/arm64/boot/dts/st/stm32mp235f-dk.dts
@@ -594,6 +594,7 @@ &sdmmc3 {
 	bus-width = <4>;
 	vmmc-supply = <&scmi_v3v3>;
 	mmc-pwrseq = <&wifi_pwrseq>;
+	cap-power-off-card;
 	cap-sdio-irq;
 	#address-cells = <1>;
 	#size-cells = <0>;
diff --git a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
index 88bafebefe06..96bd5ed289ff 100644
--- a/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-dk.dts
@@ -737,6 +737,7 @@ &sdmmc3 {
 	bus-width = <4>;
 	vmmc-supply = <&scmi_v3v3>;
 	mmc-pwrseq = <&wifi_pwrseq>;
+	cap-power-off-card;
 	cap-sdio-irq;
 	#address-cells = <1>;
 	#size-cells = <0>;
-- 
2.53.0

